specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2661 @@
+          "type": "string"
+        },
+        "value": {
+          "description": "List of Caches.",
+          "items": {
+            "$ref": "#/definitions/amlFilesystem"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "amlFilesystemIdentity": {
+      "description": "amlFilesystem identity properties.",
+      "properties": {
+        "principalId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The principal id of the amlFilesystem."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant id associated with the amlFilesystem."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of identity used for the amlFilesystem",
+          "enum": [
+            "SystemAssigned",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "amlFilesystemIdentityType",
+            "modelAsString": false
+          }
+        }
+      }
+    },
+    "amlFilesystemMaintenanceStatus": {
+      "description": "Properties describing the software upgrade state of the amlFilesystem.",
+      "properties": {
+        "currentSoftwareVersion": {
+          "description": "Version string of the software currently installed on this amlFilesystem.",
+          "type": "string",
+          "readOnly": true
+        },
+        "softwareUpdateStatus": {
+          "description": "True if there is a software update ready to install on this amlFilesystem. The firmware will automatically be installed after firmwareUpdateDeadline if not triggered earlier via the upgrade operation.",
+          "enum": [
+            "available",
+            "unavailable"
+          ],
+          "x-ms-enum": {
+            "name": "FirmwareStatusType",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "type": "string"
+        }
+      },
+      "readOnly" : true,
+      "type": "object"
+    },
+    "amlFilesystemHealth": {
+      "description": "An indication of amlFilesystem health. Gives more information about health than just that related to provisioning.",
+      "properties": {
+        "state": {
+          "description": "List of amlFilesystem health states.",
+          "enum": [
+            "Unknown",
+            "Healthy",
+            "Down",
+            "Transitioning",
+            "Maintenance"
+          ],
+          "x-ms-enum": {
+            "name": "amlFilesystemHealthStateType",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "statusDescription": {
+          "description": "Describes of state.",
+          "type": "string"
+        }
+      }
+    },
+    "amlFilesystemEncryptionSettings": {
+      "description": "amlFilesystem encryption settings.",
+      "properties": {
+        "keyEncryptionKey": {
```

Comment:
This is a KEK with no key to encrypt. See related comments; I think we need a secret either in addition to or instead of this.

Reply:
Can we reuse CRP definitions? Does it make sense to do so here?

Reply:
I thought the KeyEncryptionKey in the amlFilesystemsEncryptionSettings would stay as is to match the cacheEncryptionSettings.
CRP’s definitions may be outdated and must be there for backward compatibility.
I found Tim's notes from 8/21 RE: Resolve Swagger Comments that seem to cover that discussion.
Let me know if we still need to change this.  I left this alone since I thought it was resolved and staying as is.